From e408a4bbbeed35ff63e13bb0f41d39982a61648c Mon Sep 17 00:00:00 2001
From: Kihwal Lee <kihwal@apache.org>
Date: Mon, 5 Jun 2017 15:28:01 -0500
Subject: [PATCH 2554/2748] HDFS-10816.
 TestComputeInvalidateWork#testDatanodeReRegistration
 fails due to race between test and replication
 monitor. Contributed by Eric Badger.

(cherry picked from commit e2ea6012311c927cd7bc1524f811411fda4fd6cc)
(cherry picked from commit 1d3fa8701151e24e5ab680dd7e7152f8352252fb)

Change-Id: Icaf4dfc79197e52c314d6212e4989d1d3cb9f6d3
---
 .../blockmanagement/TestComputeInvalidateWork.java |    1 +
 1 file changed, 1 insertion(+)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/blockmanagement/TestComputeInvalidateWork.java b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/blockmanagement/TestComputeInvalidateWork.java
index fecca4e..94600ae 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/blockmanagement/TestComputeInvalidateWork.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/blockmanagement/TestComputeInvalidateWork.java
@@ -64,6 +64,7 @@ public void setup() throws Exception {
     namesystem = cluster.getNamesystem();
     bm = namesystem.getBlockManager();
     nodes = bm.getDatanodeManager().getHeartbeatManager().getDatanodes();
+    BlockManagerTestUtil.stopReplicationThread(bm);
     assertEquals(nodes.length, NUM_OF_DATANODES);
   }
 
-- 
1.7.9.5

